c7n/resources/ec2.py

```diff
@@ -606,6 +606,86 @@ def process_volume_set(self, resource):
                     response['SnapshotId']],
                 Tags=tags)
 
+@actions.register('rename-tag')
+class RenameTag(BaseAction):
+
+    schema = type_schema(
+        'rename-tag',
+        map={
+            'type': 'object',
+            'properties': {
+                'old_key': 'string',
+                'new_key': 'string'},
+            'required': ['old_key', 'new_key']})
+
+    def delete_tag(self, client, ids, key, value):
+        client.delete_tags(
+            Resources=ids,
+            Tags=[{'Key': key, 'Value': value}])
+
+    def create_tag(self, client, ids, key, value):
+        client.create_tags(
+            Resources=ids,
+            Tags=[{'Key': key, 'Value': value}])
+
+    def process_ec2(self, source_tag, source_value, resources):
+        """
+        Move source tag value to destination tag value
+
+        - Collect value from old tag
+        - Delete old tag
+        - Create new tag & assign stored value
+        """
+        self.log.info("Renaming tag on %s instances" % (len(resources)))
+        tag_map = self.data.get('map', {})
+        for t in tag_map:
```

Comment:
why is map being referenced multiple times in this class, when its not something a user can set per the schema... ie its always empty.